### PR TITLE
Added lib defs for apollo-link-http@1.2.x

### PR DESCRIPTION
**Pull request by ctgardner**
Seeking clarity about a few things before merging. (Sorry. First-time lib def author/contributor.)

### CODE REVIEW

File: definitions/npm/apollo-link-http_v1.2.x/flow_v0.28.x-/apollo-link-http_v1.2.x.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+declare module "apollo-link-http" {
+  declare export interface FetchOptions {
+    uri?: string;
+    //todo change to type of Global['fetch']
+    fetch?: any;
+    includeExtensions?: boolean;
+    credentials?: string;
+    headers?: any;
+    fetchOptions?: any;
```
Comment (ctgardner) [inline definitions/npm/apollo-link-http_v1.2.x/flow_v0.28.x-/apollo-link-http_v1.2.x.js]: Should I use `mixed` here instead (for both `headers` and `fetchOptions`)?

File: definitions/npm/apollo-link-http_v1.2.x/flow_v0.28.x-/apollo-link-http_v1.2.x.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+declare module "apollo-link-http" {
+  declare export interface FetchOptions {
+    uri?: string;
+    //todo change to type of Global['fetch']
+    fetch?: any;
```
Comment (ctgardner) [inline definitions/npm/apollo-link-http_v1.2.x/flow_v0.28.x-/apollo-link-http_v1.2.x.js]: What is `Global['fetch']`? What is the type of `Global['fetch']`? (This is how the value is assigned in the source.)

Is it OK to use `any` here? Or `mixed`?

File: definitions/npm/apollo-link-http_v1.2.x/flow_v0.28.x-/apollo-link-http_v1.2.x.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    headers?: any;
+    fetchOptions?: any;
+  }
+
+  //todo change return type to ApolloLink from 'apollo-link'
+  declare export function createHttpLink(opts: FetchOptions): any;
+
+  declare export class HttpLink {
+    //todo change type to RequestHandler from 'apollo-link'
+    requester: any;
```
Comment (ctgardner) [inline definitions/npm/apollo-link-http_v1.2.x/flow_v0.28.x-/apollo-link-http_v1.2.x.js]: The type of `requester` is `RequestHandler` from the `apollo-link` package. What is the appropriate way to declare this? (Keep in mind, `apollo-link` might not have lib defs either.)

File: definitions/npm/apollo-link-http_v1.2.x/flow_v0.28.x-/apollo-link-http_v1.2.x.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    //todo change to type of Global['fetch']
+    fetch?: any;
+    includeExtensions?: boolean;
+    credentials?: string;
+    headers?: any;
+    fetchOptions?: any;
+  }
+
+  //todo change return type to ApolloLink from 'apollo-link'
+  declare export function createHttpLink(opts: FetchOptions): any;
```
Comment (ctgardner) [inline definitions/npm/apollo-link-http_v1.2.x/flow_v0.28.x-/apollo-link-http_v1.2.x.js]: Despite the name, the `createHttpLink` function seems to return an `ApolloLink` which is from the `apollo-link` package. What is the appropriate way to declare this? (Keep in mind, `apollo-link` might not have lib defs either.)

File: definitions/npm/apollo-link-http_v1.2.x/flow_v0.28.x-/apollo-link-http_v1.2.x.js (hunk 3 above)
Comment (MichaelDeBoey) [reply definitions/npm/apollo-link-http_v1.2.x/flow_v0.28.x-/apollo-link-http_v1.2.x.js]: Dependencies between libdefs will probably been handled in the coming `3.0` release, which is currently discussed in #1494.

For now I think it's best to just duplicate the `RequestHandler` inside this libdef (like it's done in `react-router-dom` & `react-router-native` for the duplicate types from `react-router`) and add a `// TODO:` to delete this in the future when dependencies between libdefs are handled.

File: definitions/npm/apollo-link-http_v1.2.x/flow_v0.28.x-/apollo-link-http_v1.2.x.js (hunk 4 above)
Comment (MichaelDeBoey) [reply definitions/npm/apollo-link-http_v1.2.x/flow_v0.28.x-/apollo-link-http_v1.2.x.js]: Same as in my last comment: [#1552 (discussion comment)](https://github.com/flowtype/flow-typed/pull/1552#discussion_r152002379)

File: definitions/npm/apollo-link-http_v1.2.x/flow_v0.28.x-/apollo-link-http_v1.2.x.js (hunk 1 above)
Comment (GAntoine) [reply definitions/npm/apollo-link-http_v1.2.x/flow_v0.28.x-/apollo-link-http_v1.2.x.js]: `any` is fine, as it fetches the `ts` definition.

File: definitions/npm/apollo-link-http_v1.2.x/flow_v0.28.x-/apollo-link-http_v1.2.x.js (hunk 2 above)
Comment (GAntoine) [reply definitions/npm/apollo-link-http_v1.2.x/flow_v0.28.x-/apollo-link-http_v1.2.x.js]: `any` is fine, as it matches the `ts` definition.

**Pull request closed without merging**